vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -82,10 +83,41 @@ type controllerFetcher struct {
 	scaleNamespacer scale.ScalesGetter
 	mapper          apimeta.RESTMapper
 	informersMap    map[wellKnownController]cache.SharedIndexInformer
+	controllerCache controllerCacheStorage
+}
+
+func (f *controllerFetcher) periodicallyRemoveExpired(ctx context.Context, period time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(period):
+			f.controllerCache.RemoveExpired()
```

Comment:
Maybe it would be useful here to log start & finish (or inside RemoveExpired, to get the number of entries removed).

Reply:
Done.